website/content/docs/external_plugins/_index.md

```diff
@@ -141,6 +152,34 @@ Below is an example of valid `list` output:
 ]
 ```
 
+If you're able to pre-fetch a method's result as part of the `list` method, then you can include the result as a tuple of `[<method>, <result>]` in the `methods` array. Pre-fetching is a useful way to avoid unnecessary plugin script invocations. Below is an example that includes pre-fetched method results. Note that when `read` content is provided in this manner, the size of that content will be automatically populated in `attributes`.
```

Comment:
What I had before was
```
If your plugin knows some of the hierarchy and wants to describe it now rather than relying on more calls to `list` or `read` methods, it can do so by describing that hierarchy in the `methods` property of any attribute.

Instead of providing a list of methods to `methods`, provide an object. The keys of that object are supported methods on the entry. If the key maps to `null`, then the method is supported and will invoke your script again for that method (just like providing the method name in an array). The following alternative values may be provided
- for the `list` method, an object corresponding to what would be returned by invoking the `list` method on the script
- for the `read` method, a string containing the entry's contents

Below is an example JSON object declaring a static directory that contains known files and content, but may also support streaming new updates dynamically (by invoking the `stream` method on the script).
```

Not sure what you were referring to.

Reply:
Hmm, looks like the example here corresponds to the `static directory` case so may be worth mentioning that.

Was thinking a `Prefetch Scenarios` section could be useful so we could also describe Corey's use-case with the hash, but maybe that's too early.

Reply:
I added more explanation of the example. Not ready to add scenarios.